scripts-dev/complement.sh

```diff
@@ -181,24 +181,40 @@ if [ -z "$skip_docker_build" ]; then
 
         # Build the base Synapse image from the local checkout
         echo_if_github "::group::Build Docker image: matrixdotorg/synapse"
-        $CONTAINER_RUNTIME build -t matrixdotorg/synapse \
-        --build-arg TEST_ONLY_SKIP_DEP_HASH_VERIFICATION \
-        --build-arg TEST_ONLY_IGNORE_POETRY_LOCKFILE \
-        -f "docker/Dockerfile" .
+
+        # Tag local builds with a dummy registry so that later builds
+        # may reference them instead of pulling from a remote registry
+        LOCAL_REGISTRY=localhost:5000
```

Comment:
```suggestion
        LOCAL_REGISTRY=localhost
```

I'd rather not use a port number if it's a dummy, as this is confusing.

Not necessarily sure this change is necessary as it already used to use the intermediate builds without pulling, but I guess it's fine.

Reply:
Ok, I'll revert all of the local registry mangling in favour of keeping it in https://github.com/element-hq/synapse/pull/18210.

Reply:
5d9fe30